pandas/tests/frame/conftest.py

```diff
@@ -70,9 +82,10 @@ def mixed_float_frame():
     Columns are ['A', 'B', 'C', 'D'].
     """
     df = DataFrame(tm.getSeriesData())
-    df.A = df.A.astype('float16')
+    df.A = df.A.astype('float32')
```

Comment:
What's the reasoning for changing these?

Reply:
See OP:
> In translating the quasi-fixtures from `TestData` to `conftest` in #22236, I sorted the dtypes for the columns of `mixed_float_frame` and `mixed_int_frame`, which turns out to have been a mistake. This is reverted here to be a true translation of the attribute of `TestData`. Otherwise, tests in the newly fixturized `test_arithmetic.py` would fail.